From: Jose Castillo <jcastillo@redhat.com>
Date: Fri, 29 Jan 2016 14:39:26 +0000
Subject: bnx2fc: Show information about log levels in 'modinfo'
References: bsc#966327,FATE#320151
Patch-mainline: v4.6-rc1
Git-commit: d4faee084f04ce2d9d97502b3471d77e6fea9d4b

This patch adds the information of the different values that can be used
in the module parameter 'debug_logging', as it is shown below:

$ modinfo bnx2fc
[...]
parm:           debug_logging:Option to enable extended logging,
		Default is 0 - no logging.
		0x01 - SCSI cmd error, cleanup.
		0x02 - Session setup, cleanup, etc.
		0x04 - lport events, link, mtu, etc.
		0x08 - ELS logs.
		0x10 - fcoe L2 fame related logs.
		0xff - LOG all messages. (int)

Signed-off-by: Jose Castillo <jcastillo@redhat.com>
Acked-by: Chad Dupuis <chad.dupuis@qlogic.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/bnx2fc/bnx2fc_fcoe.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/scsi/bnx2fc/bnx2fc_fcoe.c b/drivers/scsi/bnx2fc/bnx2fc_fcoe.c
index 8c520f2..0bf9459 100644
--- a/drivers/scsi/bnx2fc/bnx2fc_fcoe.c
+++ b/drivers/scsi/bnx2fc/bnx2fc_fcoe.c
@@ -97,6 +97,15 @@ static void __exit bnx2fc_mod_exit(void);
 
 unsigned int bnx2fc_debug_level;
 module_param_named(debug_logging, bnx2fc_debug_level, int, S_IRUGO|S_IWUSR);
+MODULE_PARM_DESC(debug_logging,
+		"Option to enable extended logging,\n"
+		"\t\tDefault is 0 - no logging.\n"
+		"\t\t0x01 - SCSI cmd error, cleanup.\n"
+		"\t\t0x02 - Session setup, cleanup, etc.\n"
+		"\t\t0x04 - lport events, link, mtu, etc.\n"
+		"\t\t0x08 - ELS logs.\n"
+		"\t\t0x10 - fcoe L2 fame related logs.\n"
+		"\t\t0xff - LOG all messages.");
 
 static int bnx2fc_cpu_callback(struct notifier_block *nfb,
 			     unsigned long action, void *hcpu);
-- 
2.6.2

